extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,86 @@
+import json
+from packageurl import PackageURL
```

Comment:
How is this dependency supposed to be installed? any information this is a requirement? 

If you use external dependencies in the first place, then you really should consider using https://pypi.org/project/cyclonedx-python-lib/ to generate the CycloneDX document, instead of crafting the document manually.

Reply:
I did not find in the docu how to specify requirements for conan extensions. How to do that? Can someone help here?

Reply:
@memsharded (how) can a conan2 extension add python dependencvies?

Reply:
We are very strict about adding dependencies, this should be avoid as much as possible https://github.com/conan-io/conan/pull/14405#discussion_r1280335011

Reply:
This is fine for the (core) conan package. However, is this also as strict for conan-extensions?

Reply:
so if no 3rd party libs can be used, then the crafting of CycloneDX JSON, crafing of PURLs and other things need to be done by hand or re-implemented here, right? 

should he roll back, @prince-chrismc ?
well, if it would be not possible to use external libraries , then 

- crafting proper CycloneDX JSON is not that hard, as @hedtke did it before switching to the CycloneDX-lib via https://github.com/conan-io/conan-extensions/pull/66/commits/31b701ce9bc9f590a64a27830289deb006745bf6
- Crafting a valid CycloneDX license should be possible by checking whether a license is in the list of supported SPDX IDs (see <https://cyclonedx.org/schema/spdx.schema.json>). If the license is known, then [SPDX license ID](https://cyclonedx.org/docs/1.5/json/#tab-pane_components_items_licenses_oneOf_i0_items_license_oneOf_i0) can be used, else a [named license](https://cyclonedx.org/docs/1.5/json/#tab-pane_components_items_licenses_oneOf_i0_items_license_oneOf_i1) shall be used instead.
  Something like 
  ```py
  # a dict of all SPDX licenses supported by CycloneDX, keys are the values just lowercased - can also created dynamically like dict([k.lower(), k] for k in {... all values here ...})
  _cdx_licenses_lowercased = {'0bsd':'0BSD', 'aal':'AAL', '...': '... and all others ...'}  # all the values from https://cyclonedx.org/schema/spdx.schema.json 

  make_cdx_license(license: str) -> dict: 
    licenseId = _cdx_licenses_lowercased.get(license.lower(), None)
    return {'id':licenseId} if licenseId else {'name':licenseId}
  ```
- Crafting a PURL is not that hard either, it just needs carefully url-encoding several parts of the string, as described here https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component and here https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#character-encoding

Reply:
@prince-chrismc really no external libs in extensions, too?

Reply:
At the moment, the ``conan-extensions`` doesn't have a mechanism to install ``pip`` dependencies in the current Python environment automatically, so at the moment this should be something done when installing ``pip install conan package-url ...`` as necessary. 

Note that adding them to a ``requirements.txt`` in this current repo is pointless, as things here are intended to be installed with ``conan config install``.

We can think in the future of possible ways to automate this, at the moment I'd recommend to ``try-except`` the import with a clear message ``The sbom extension needs the 'package-url' package, please 'pip install ...'``

Reply:
I found a problem with that approach: In installed the extensions in an environment where I do not have the external libs installed. Now even running a non-extension command triggers an Exception:

```
$ conan install .
Migration: Successfully updated settings.yml
The sbom extension needs the 'packageurl' package, please run 'pip install packageurl-python'
ERROR: Exiting with code: 1
```

Reply:
I changed the approach: Required libs are checked only then the new command is called